Upstream-Status: unsubmitted

From 05ab022edc9b28bfa85794db0ca85776f61975f5 Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Fri, 1 Jul 2022 16:12:29 +0200
Subject: [PATCH] delta-ag5648: avoid multiple definitions of mutex/mutex1

Avoid multiple global definitions of mutex and mutex1 through inclusion
of the header file by moving the definitions into the only place where
they are used, platform_lib.c.

Avoid the following build error with GCC 10+:

|     LinkingShared[release]: libonlp-x86-64-delta-ag5648::libonlp-x86-64-delta-ag5648.so
| /tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/recipe-sysroot-native/usr/bin/x86_64-poky-linux/../../libexec/x86_64-poky-linux/gcc/x86_64-poky-linux/11.3.0/ld: BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/ledi.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:166: multiple definition of `mutex1'; BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/fani.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:166: first defined here
| /tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/recipe-sysroot-native/usr/bin/x86_64-poky-linux/../../libexec/x86_64-poky-linux/gcc/x86_64-poky-linux/11.3.0/ld: BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/ledi.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:165: multiple definition of `mutex'; BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/fani.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:165: first defined here
| /tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/recipe-sysroot-native/usr/bin/x86_64-poky-linux/../../libexec/x86_64-poky-linux/gcc/x86_64-poky-linux/11.3.0/ld: BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:166: multiple definition of `mutex1'; BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/fani.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:166: first defined here
| /tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/recipe-sysroot-native/usr/bin/x86_64-poky-linux/../../libexec/x86_64-poky-linux/gcc/x86_64-poky-linux/11.3.0/ld: BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:165: multiple definition of `mutex'; BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/fani.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:165: first defined here
| /tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/recipe-sysroot-native/usr/bin/x86_64-poky-linux/../../libexec/x86_64-poky-linux/gcc/x86_64-poky-linux/11.3.0/ld: BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/psui.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:166: multiple definition of `mutex1'; BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/fani.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:166: first defined here
| /tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/recipe-sysroot-native/usr/bin/x86_64-poky-linux/../../libexec/x86_64-poky-linux/gcc/x86_64-poky-linux/11.3.0/ld: BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/psui.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:165: multiple definition of `mutex'; BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/fani.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:165: first defined here
| /tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/recipe-sysroot-native/usr/bin/x86_64-poky-linux/../../libexec/x86_64-poky-linux/gcc/x86_64-poky-linux/11.3.0/ld: BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/sfpi.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:166: multiple definition of `mutex1'; BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/fani.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:166: first defined here
| /tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/recipe-sysroot-native/usr/bin/x86_64-poky-linux/../../libexec/x86_64-poky-linux/gcc/x86_64-poky-linux/11.3.0/ld: BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/sfpi.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:165: multiple definition of `mutex'; BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/fani.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:165: first defined here
| /tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/recipe-sysroot-native/usr/bin/x86_64-poky-linux/../../libexec/x86_64-poky-linux/gcc/x86_64-poky-linux/11.3.0/ld: BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/sysi.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:166: multiple definition of `mutex1'; BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/fani.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:166: first defined here
| /tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/recipe-sysroot-native/usr/bin/x86_64-poky-linux/../../libexec/x86_64-poky-linux/gcc/x86_64-poky-linux/11.3.0/ld: BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/sysi.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:165: multiple definition of `mutex'; BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/fani.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:165: first defined here
| /tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/recipe-sysroot-native/usr/bin/x86_64-poky-linux/../../libexec/x86_64-poky-linux/gcc/x86_64-poky-linux/11.3.0/ld: BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/thermali.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:166: multiple definition of `mutex1'; BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/fani.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:166: first defined here
| /tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/recipe-sysroot-native/usr/bin/x86_64-poky-linux/../../libexec/x86_64-poky-linux/gcc/x86_64-poky-linux/11.3.0/ld: BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/thermali.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:165: multiple definition of `mutex'; BUILD/buster/gcc-local/obj/tmp/kirkstone/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/fani.o:/usr/src/debug/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h:165: first defined here
| collect2: error: ld returned 1 exit status

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>
---
 .../onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.c  | 3 +++
 .../onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h  | 2 --
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.c b/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.c
index d6046c6e563f..e9150b07f2ba 100755
--- a/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.c
+++ b/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.c
@@ -36,6 +36,9 @@
 #include <onlplib/mmap.h>
 #include <pthread.h>
 
+pthread_mutex_t mutex;
+pthread_mutex_t mutex1;
+
 int dni_i2c_read_attribute_binary(char *filename, char *buffer, int buf_size, int data_len)
 {
     int fd;
diff --git a/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h b/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h
index 80792c6224d1..f686ae736fde 100755
--- a/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h
+++ b/packages/platforms/delta/x86-64/ag5648/onlp/builds/x86_64_delta_ag5648/module/src/platform_lib.h
@@ -162,8 +162,6 @@ typedef struct mux_info_s
     uint32_t flags;
 }mux_info_t;
 
-pthread_mutex_t mutex;
-pthread_mutex_t mutex1;
 int dni_i2c_lock_read(mux_info_t * mux_info, dev_info_t * dev_info);
 int dni_i2c_lock_write(mux_info_t * mux_info, dev_info_t * dev_info);
 int dni_i2c_lock_read_attribute(mux_info_t * mux_info, char * fullpath);
-- 
2.36.1

